packages/saltcorn-data/base-plugin/viewtemplates/filter.js

```diff
@@ -495,17 +528,83 @@ const eq_string = (x, y) => `${x}` === `${y}`;
 const run_action = async (
   table_id,
   viewname,
-  { columns, layout },
+  config,
   body,
   { req, res },
   { actionQuery }
 ) => {
-  const result = await actionQuery();
+  const table = Table.findOne({ id: table_id });
+  if (!table)
+    throw new InvalidConfiguration(
+      `View '${viewname}:run_action' incorrectly configured: ` +
+        `Unable to find table with id '${table_id}'`
+    );
+  const state = req?.query
+    ? readState(removeEmptyStrings(req.query), table.getFields(), req)
+    : {};
+  const result = await actionQuery(state, body?.rndid);
   if (result.json.error) {
     Crash.create({ message: result.json.error, stack: "" }, req);
   }
   return result;
 };
+
+/**
+ * combine multiple action results into one object
+ * for 'reload_page, goto, popup' take the first
+ * all other types are combined into arrays
+ * @param results array of action results
+ */
+const combineResults = (results) => {
+  const messageLimit = 5;
+  const downloadLimit = 5;
+  let numMsgs = 0,
+    suppressedMsgs = 0,
+    suppressedErrors = 0;
+  let numDownloads = 0,
+    suppressedDownloads = 0;
+  const result = { json: { success: "ok" } };
+  const initOrPush = (newElement, memberName) => {
+    if (result.json[memberName]) result.json[memberName].push(newElement);
+    else result.json[memberName] = [newElement];
+  };
+  const initOnce = (newElement, memberName) => {
+    if (typeof result[memberName] === "undefined")
+      result.json[memberName] = newElement;
+  };
+  for (const result of results) {
+    if (result.reload_page) initOnce(result.reload_page, "reload_page");
```

Comment:
before this line insert:

`if(!result) continue;`

not all actions will return an object and then it gives error

Reply:
fixed